src/meta-srv/src/procedure/alter_table.rs

```diff
@@ -79,41 +79,65 @@ impl AlterTableProcedure {
 
         let manager = &self.context.table_metadata_manager;
 
+        // (comparing_key, expected)
+        let mut compares = Vec::default();
+
         if let AlterKind::RenameTable { new_table_name } = &request.alter_kind {
-            let exist = manager
-                .table_name_manager()
-                .exist(TableNameKey::new(
-                    &request.catalog_name,
-                    &request.schema_name,
-                    new_table_name,
-                ))
-                .await
-                .context(TableMetadataManagerSnafu)?;
-
-            ensure!(
-                !exist,
-                error::TableAlreadyExistsSnafu {
-                    table_name: request.table_ref().to_string()
-                }
-            )
-        }
+            // should not exist
+            compares.push((
+                TableNameKey::new(&request.catalog_name, &request.schema_name, new_table_name),
+                false,
+            ));
+        };
 
-        let exist = manager
-            .table_name_manager()
-            .exist(TableNameKey::new(
+        // should exist
+        compares.push((
+            TableNameKey::new(
                 &request.catalog_name,
                 &request.schema_name,
                 &request.table_name,
-            ))
+            ),
+            true,
+        ));
+
+        let (compare_keys, handlers): (Vec<_>, Vec<_>) = compares
+            .into_iter()
+            .enumerate()
+            .map(|(idx, (key, expected))| {
+                (key, move |exists: &Vec<bool>| -> Result<()> {
+                    if expected {
+                        ensure!(
+                            exists[idx],
+                            error::TableNotFoundSnafu {
+                                name: request.table_ref().to_string()
+                            }
+                        );
+                    } else {
+                        ensure!(
+                            !exists[idx],
+                            error::TableAlreadyExistsSnafu {
+                                table_name: request.table_ref().to_string()
+                            }
+                        );
+                    }
+
+                    Ok(())
+                })
+            })
+            .collect::<Vec<_>>()
+            .into_iter()
+            .unzip();
+
+        let exists = manager
+            .table_name_manager()
+            .batch_exists(compare_keys)
             .await
-            .context(TableMetadataManagerSnafu)?;
+            .context(error::TableMetadataManagerSnafu)?;
+
+        for handler in handlers {
+            handler(&exists)?;
+        }
```

Comment:
Apologies for not thoroughly reviewing the code yesterday. I misunderstood the requirement and believed it was to ensure everything exists. This change is overly complex. Let's revert back to the original code. ^_^